apis/goharbor.io/v1beta1/harbor_configuration.go

```diff
@@ -32,233 +32,228 @@ type HarborConfiguration struct {
+	ProjectCreationRestriction string `json:"projectCreationRestriction,omitempty" yaml:"project_creation_restriction,omitempty"`
 	// The flag to indicate whether Harbor is in readonly mode.
 	// +kubebuilder:validation:Optional
-	ReadOnly *bool `json:"read_only,omitempty"`
+	ReadOnly *bool `json:"readOnly,omitempty" yaml:"read_only,omitempty"`
 	// Whether the Harbor instance supports self-registration.  If it''s set to false, admin need to add user to the instance.
 	// +kubebuilder:validation:Optional
-	SelfRegistration *bool `json:"self_registration,omitempty"`
+	SelfRegistration *bool `json:"selfRegistration,omitempty" yaml:"self_registration,omitempty"`
 	// The expiration time of the token for internal Registry, in minutes.
 	// +kubebuilder:validation:Optional
-	TokenExpiration int `json:"token_expiration,omitempty"`
+	TokenExpiration int `json:"tokenExpiration,omitempty" yaml:"token_expiration,omitempty"`
 	// HttpAuthproxy related configurations.
 	// +kubebuilder:validation:Optional
-	HarborConfigurationHTTPAuthProxySpec `json:",inline"`
+	HarborConfigurationHTTPAuthProxy `json:",inline" yaml:",inline"`
 	// Uaa related configurations.
 	// +kubebuilder:validation:Optional
-	HarborConfigurationUaaSpec `json:",inline"`
+	HarborConfigurationUaa `json:",inline" yaml:",inline"`
 	// Oidc related configurations.
 	// +kubebuilder:validation:Optional
-	HarborConfigurationOidcSpec `json:",inline"`
+	HarborConfigurationOidc `json:",inline" yaml:",inline"`
 	// The robot account token duration in days.
 	// +kubebuilder:validation:Optional
-	RobotTokenDuration int `json:"robot_token_duration,omitempty"`
+	RobotTokenDuration int `json:"robotTokenDuration,omitempty" yaml:"robot_token_duration,omitempty"`
 	// The rebot account name prefix.
 	// +kubebuilder:validation:Optional
-	RobotNamePrefix string `json:"robot_name_prefix,omitempty"`
+	RobotNamePrefix string `json:"robotNamePrefix,omitempty" yaml:"robot_name_prefix,omitempty"`
 	// Enable notification.
 	// +kubebuilder:validation:Optional
-	NotificationEnable *bool `json:"notification_enable,omitempty"`
+	NotificationEnable *bool `json:"notificationEnable,omitempty" yaml:"notification_enable,omitempty"`
 	// Enable quota per project.
 	// +kubebuilder:validation:Optional
-	QuotaPerProjectEnable *bool `json:"quota_per_project_enable,omitempty"`
+	QuotaPerProjectEnable *bool `json:"quotaPerProjectEnable,omitempty" yaml:"quota_per_project_enable,omitempty"`
 	// The storage quota per project.
 	// +kubebuilder:validation:Optional
-	StoragePerProject int `json:"storage_per_project,omitempty"`
+	StoragePerProject int `json:"storagePerProject,omitempty" yaml:"storage_per_project,omitempty"`
 }
 
 // ToJSON converts configuration spec to json payload.
-func (h HarborConfigurationSpec) ToJSON() ([]byte, error) {
-	jsonData, err := json.Marshal(h)
+func (h HarborConfigurationModel) ToJSON() ([]byte, error) {
+	data, err := goyaml.Marshal(h)
```

Comment:
May `apiextensionsv1.JSON` also be parsed correctly?

Reply:
Yes, tested.